tests/test_unittest/test_stencil_builder.py

```diff
@@ -124,3 +124,32 @@ def test_regression_run_analysis_twice(tmp_path):
     ir = builder.implementation_ir
     # this raises an error if the analysis pipeline is reevaluated:
     assert ir is builder.implementation_ir
+
+
+def test_deferred_generate():
+    def build_function(builder):
+        return builder.backend.generate()
+
+    def defer_function(builder):
+        build_function.builder = builder
```

Comment:
This looks like it might be a leftover from some previous state of the code, since it has no effect.

If you want to lock the build function in on a specific builder this way, it would have to be
```python
def build_function():
    return self.builder.backend.generate()
```

This would not allow you to reuse the build function for different stencils though.

That could be achieved with
```python
from functools import partial

def defer_function(builder):
    return partial(build_function, builder)
```

which returns a `partial` function object that with a reference to the function (`.func`) as well as the builder (`.args`) and can be called without arguments.

For this test the line could also just be dropped but I prefer when tests also represent sane usage.

Reply:
Thank you for the feedback. The idea is that the `defer_function` returns a callable object to `stencil_class` that can later build the stencil. When the class is later instantiated, the defer function generates the stencil object using the memoized builder. My apologies for the confusion.

I shall try achieving this same effect with `LazyStencil`, did not get to that yesterday. Thank you again!